src/test/regress/sql/multi_behavioral_analytics_basics.sql

```diff
@@ -476,3 +476,117 @@ FROM
   ORDER BY 1, 2;
 
 SELECT count(*), count(DISTINCT user_id), avg(user_id), avg(value_1_agg) FROM agg_results;
+
+-- test order by when pushing down avg
+SELECT user_id, avg(value_1)
+FROM users_table
+GROUP BY user_id
+ORDER BY avg(value_1) DESC
+LIMIT 5;
+
+SELECT user_id, avg(value_1)
+FROM users_table
+GROUP BY user_id
+ORDER BY avg(value_1) DESC
+LIMIT 1;
+
+SELECT user_id, avg(value_1) + 1
+FROM users_table
+GROUP BY user_id
+ORDER BY avg(value_1) + 1 DESC
+LIMIT 1;
+
+SELECT user_id, avg(value_1)
+FROM users_table
+GROUP BY user_id
+ORDER BY avg(value_1) + 1 DESC
+LIMIT 1;
+
+SELECT user_id, avg(value_1) + sum(value_2)
+FROM users_table
+GROUP BY user_id
+ORDER BY 2  DESC
+LIMIT 1;
+
+SELECT user_id, avg(value_1) + count(value_2)
+FROM users_table
+GROUP BY user_id
+ORDER BY 2  DESC;
+
+SELECT user_id, avg(value_1) + count(value_2)
+FROM users_table
+GROUP BY user_id
+ORDER BY 2  DESC
+LIMIT 1;
+
+SELECT user_id, sum(value_1) + sum(value_2)
+FROM users_table
+GROUP BY user_id
+ORDER BY 2  DESC
+LIMIT 1;
+
+SELECT user_id, sum(value_1) + sum(value_2)
+FROM users_table
+GROUP BY user_id
+ORDER BY sum(value_2)  DESC
+LIMIT 1;
+
+SELECT user_id, (100 / max(value_1))
```

Comment:
I think we could consider adding one test with multiple target entries in the ORDER BY that are modified. We seem to have only one for all tests?

Also, maybe a test with a join such that each element belonging to a different table?

Reply:
One more test idea: Aggregate inside distinct: `select distinct user_id, (avg (value_1)) FROM users_table GROUP BY user_id ORDER BY 2 DESC, 1 DESC LIMIT 10;`
(Is there a way to pushdown `DISTINCT` with aggragates)

Reply:
added